pkg/kn/flags/podspec.go

```diff
@@ -177,6 +180,18 @@ func (p *PodSpecFlags) AddFlags(flagset *pflag.FlagSet) []string {
 			"Example: --containers ./containers.yaml or --containers -.")
 	flagNames = append(flagNames, "containers")
 
+	// Probes
+	commonProbeDescription := "Supported probe types are HTTGet, Exec and TCPSocket. " +
+		"Format: [http,https]:host:port:path;<common_opts>, exec:cmd,cmd,...;<common_opts>, tcp:host:port;<common_opts>. " +
+		"Common opts (comma separated, case insensitive): InitialDelaySeconds=<int_value>, FailureThreshold=<int_value>, " +
+		"SuccessThreshold=<int_value>, PeriodSeconds=<int_value>, TimeoutSeconds=<int_value>"
+	flagset.StringVarP(&p.LivenessProbe, "probe-liveness", "", "", "Add liveness probe to Service deployment. "+
+		commonProbeDescription)
+	flagNames = append(flagNames, "probe-liveness")
+	flagset.StringVarP(&p.ReadinessProbe, "probe-readiness", "", "", "Add readiness probe to Service deployment. "+
+		commonProbeDescription)
+	flagNames = append(flagNames, "probe-readiness")
```

Comment:
at some point we should also add a `probe-startup` for the startup probe (but not sure whether startup probes are supported by knative, we should check this)

Reply:
That's not supported, as far as I've seen. 

Per: https://github.com/knative/serving/blob/main/pkg/apis/serving/k8s_validation.go#L513-L520

```
error: services.serving.knative.dev "hello" could not be patched: admission webhook "validation.webhook.serving.knative.dev" denied the request: validation failed: must not set the field(s): spec.template.spec.containers[0].startupProbe
```

Reply:
The flag should be a quick additions once enabled on the Serving side.